src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -225,19 +230,30 @@ public class TestConnectionCommand : PSCmdlet, IDisposable
         [ValidateRange(0, 65535)]
         [Parameter(Mandatory = true, ParameterSetName = TcpPortParameterSet)]
         public int TcpPort { get; set; }
-
+        
         #endregion Parameters
 
         /// <summary>
         /// BeginProcessing implementation for TestConnectionCommand.
+        /// Sets Count for different types of tests unless specified explicitly.
         /// </summary>
         protected override void BeginProcessing()
         {
-            switch (ParameterSetName)
+            if (Count == -1)
```

Comment:
```suggestion
            if (MyInvocation.BoundParameters.ContainsKey(name of(Count))
```

Reply:
Please use `switch (ParameterSetName)` and do initializations for every paramter set in `case`.